extensions/amp-access/0.1/test/test-login-dialog.js

```diff
@@ -22,10 +22,130 @@ const RETURN_URL_ESC = encodeURIComponent('http://localhost:8000/extensions' +
     encodeURIComponent('http://localhost:8000/test-login-dialog'));
 
 
-describe('LoginDialog', () => {
+describe('ViewerLoginDialog', () => {
+
+  let sandbox;
+  let viewer;
+  let windowApi;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    clock = sandbox.useFakeTimers();
+
+    viewer = {
+      getParam: param => {
+        if (param == 'dialog') {
+          return '1';
+        }
+        return null;
+      },
+      sendMessage: () => {},
+    };
+
+    windowApi = {
+      services: {
+        'viewer': {obj: viewer},
+      },
+      screen: {width: 1000, height: 1000},
+      open: () => {
+        throw new Error('Not allowed');
+      },
+      addEventListener: () => {
+        throw new Error('Not allowed');
+      },
+      setTimeout: () => {
+        throw new Error('Not allowed');
+      },
+      setInterval: () => {
+        throw new Error('Not allowed');
+      },
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    sandbox = null;
+  });
+
+  it('should delegate to viewer with url', () => {
+    const stub = sandbox.stub(viewer, 'sendMessage',
+        () => Promise.resolve('#success=yes'));
+    return openLoginDialog(windowApi, 'http://acme.com/login').then(res => {
+      expect(res).to.equal('#success=yes');
```

Comment:
Not expected, but allowed/tolerated. So, either/or. But from documentation point of view I usually like to include it in tests/examples since it makes it very clear what part of URL is of interest.